app/src/split/serial/service.c

```diff
@@ -0,0 +1,106 @@
+/*
+* Copyright (c) 2022 The ZMK Contributors
+*
+* SPDX-License-Identifier: MIT
+*/
+
+#include <zmk/split/common.h>
+#include <sys/util.h>
+#include <sys/crc.h>
+#include <init.h>
+
+#include <device.h>
+#include <drivers/uart.h>
+
+#include <logging/log.h>
+
+LOG_MODULE_DECLARE(zmk, CONFIG_ZMK_LOG_LEVEL);
+
+#include <zmk/matrix.h>
+
+#if DT_HAS_CHOSEN(zmk_split_serial)
+
+#define UART_NODE1 DT_CHOSEN(zmk_split_serial)
+const struct device *serial_dev = DEVICE_DT_GET(UART_NODE1);
+static int uart_ready = 0;
+
+static uint8_t position_state[SPLIT_DATA_LEN];
+
+K_THREAD_STACK_DEFINE(service_q_stack, CONFIG_ZMK_SPLIT_SERIAL_PERIPHERAL_STACK_SIZE);
+
+struct k_work_q service_work_q;
+
+K_MSGQ_DEFINE(position_state_msgq, sizeof(char[SPLIT_DATA_LEN]),
+              CONFIG_ZMK_SPLIT_SERIAL_PERIPHERAL_POSITION_QUEUE_SIZE, 4);
+
+
+void send_data_via_uart(const struct device *dev, char *data, size_t len) {
+    if (!uart_ready) {
+        return;
+    }
+
+    for (int i=0; i < len; i++) {
+        uart_poll_out(serial_dev, data[i]);
```

Comment:
Why not `uart_tx` for the whole buffer?

Reply:
I have done the TX and RX part first using sync mode and then updated RX to async using DMA.
Will update TX also to use same apis